backends/openvino/runtime/OpenvinoBackend.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) Intel Corporation
+ *
+ * Licensed under the BSD License (the "License"); you may not use this file
+ * except in compliance with the License. See the license file in the root
+ * directory of this source tree for more details.
+ */
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+
+#include <openvino/openvino.hpp>
+
+#include <executorch/runtime/backend/interface.h>
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/evalue.h>
+#include <executorch/runtime/core/exec_aten/util/dim_order_util.h>
+#include <executorch/runtime/core/exec_aten/util/scalar_type_util.h>
+
+#include "OpenvinoBackend.hpp"
+
+using namespace std;
+using executorch::aten::ScalarType;
+using executorch::runtime::ArrayRef;
+using executorch::runtime::Backend;
+using executorch::runtime::BackendExecutionContext;
+using executorch::runtime::BackendInitContext;
+using executorch::runtime::CompileSpec;
+using executorch::runtime::DelegateHandle;
+using executorch::runtime::Error;
+using executorch::runtime::EValue;
+using executorch::runtime::FreeableBuffer;
+using executorch::runtime::MemoryAllocator;
+using executorch::runtime::Result;
+
+namespace executorch {
+namespace backends {
+namespace openvino {
+
+OpenvinoBackend::OpenvinoBackend() {
+  if (!is_available()) {
+    // ET_LOG(Error, "OpenVINO runtime is not available. Initialization
+    // failed.");
+    throw std::runtime_error("OpenVINO runtime not available");
+  }
+
+  // ET_LOG(Info, "OpenVINO runtime successfully verified and initialized.");
+}
+
+bool OpenvinoBackend::is_available() const {
+  try {
+    // Create an OpenVINO Core object to verify runtime availability
+    ov::Core core;
+
+    // Check if at least one device is available
+    auto devices = core.get_available_devices();
+    if (!devices.empty()) {
+      return true; // OpenVINO is available
+    }
+  } catch (const std::exception& e) {
+    // Log the exception if OpenVINO runtime is not available
+    ET_LOG(Error, "OpenVINO is not available: %s", e.what());
+  } catch (...) {
+    // Handle any unexpected errors
+    ET_LOG(
```

Comment:
Why are we just logging error instead of actually raising it?

Reply:
In case of an error, this function returns `false` and the caller (in `OpenvinoBackend` constructor) raises the error. Do you still suggest raising the error inside this function or keep it as it is and raise the error where the function is called?